apm-agent-core/src/main/java/co/elastic/apm/agent/impl/circuitbreaker/CircuitBreakerConfiguration.java

```diff
@@ -0,0 +1,151 @@
+            "When inactive, the agent continues polling the same monitors in order to detect whether the stress state \n" +
+            "has been relieved. If ALL monitors approve that the system/process/JVM is not under stress anymore, the \n" +
+            "agent will resume and become fully functional.")
+        .dynamic(true)
+        .buildWithDefault(false);
+
+    private final ConfigurationOption<TimeDuration> stressMonitoringInterval = TimeDurationValueConverter.durationOption("s")
+        .key("stress_monitoring_interval")
+        .tags("added[1.14.0]")
+        .tags("performance")
+        .configurationCategory(CIRCUIT_BREAKER_CATEGORY)
+        .description("The interval at which the agent polls the stress monitors. Must be at least `1s`.")
+        .addValidator(isNotInRange(TimeDuration.of("0ms"), TimeDuration.of("999ms")))
+        .dynamic(false)
+        .buildWithDefault(TimeDuration.of("5s"));
+
+    private final ConfigurationOption<Double> gcStressThreshold = ConfigurationOption.doubleOption()
+        .key("stress_monitor_gc_stress_threshold")
+        .configurationCategory(CIRCUIT_BREAKER_CATEGORY)
+        .tags("added[1.14.0]")
+        .tags("performance")
+        .description("The threshold used by the GC monitor to rely on for identifying heap stress.\n" +
+            "The same threshold will be used for all heap pools, so that if ANY has a usage percentage that crosses it, \n" +
+            "the agent will consider it as a heap stress. The GC monitor relies only on memory consumption measured \n" +
+            "after a recent GC.")
+        .dynamic(true)
+        .addValidator(isInRange(0d, 1d))
+        .buildWithDefault(0.95);
+
+    private final ConfigurationOption<Double> gcReliefThreshold = ConfigurationOption.doubleOption()
+        .key("stress_monitor_gc_relief_threshold")
+        .configurationCategory(CIRCUIT_BREAKER_CATEGORY)
+        .tags("added[1.14.0]")
+        .tags("performance")
+        .description("The threshold used by the GC monitor to rely on for identifying when the heap is not under stress .\n" +
+            "If `stress_monitor_gc_stress_threshold` has been crossed, the agent will consider it a heap-stress state. \n" +
+            "In order to determine that the stress state is over, percentage of occupied memory in ALL heap pools should \n" +
+            "be lower than this threshold. The GC monitor relies only on memory consumption measured after a recent GC.")
+        .dynamic(true)
+        .addValidator(isInRange(0d, 1d))
+        .buildWithDefault(0.75);
+
+    private final ConfigurationOption<Integer> cpuConsecutiveMeasurements = ConfigurationOption.integerOption()
+        .key("stress_monitor_cpu_num_measurements")
+        .configurationCategory(CIRCUIT_BREAKER_CATEGORY)
+        .tags("added[1.14.0]")
+        .tags("performance")
+        .description("The number of measurements required in order to determine whether the system is \n" +
```

Comment:
Is that a number of consecutive measurements ? if yes (I guess it's the case) might be relevant to make it explicit here.

Reply:
I am changing this one based on https://github.com/elastic/apm-agent-java/pull/1040#discussion_r386428151